### PR TITLE
Base image update

### PR DESCRIPTION
**Pull request by numberth**
Changed where base is initiated and added image update to base

### CODE REVIEW

File: src/basic/Entity.h [hunk 1]
```diff
@@ -55,9 +55,11 @@ class Entity {
     float getY() const{return y;} // Get y coordinate
     int getW() const{return destRect.w;} // Get width
     int getH() const{return destRect.h;} // Get height
+    void setW(int w) {destRect.w = w;} // Set width
+    void setH(int h) {destRect.h = h;} // Set height
```
Comment (JohnAgapeyev) [inline src/basic/Entity.h]: const int

File: src/buildings/Base.h [hunk 2]
```diff
@@ -5,28 +5,53 @@
...
+
 //The gab between the spawn point and base.
 static constexpr int GAP = 150;
 
 typedef std::pair<float, float> Point;
 
 class Base : public Object {
 public:
-    Base(const int32_t nid = 0, const SDL_Rect& dest = {MAP_WIDTH / 2, MAP_HEIGHT / 2, BASE_WIDTH, BASE_HEIGHT}, int health = 100);
+    Base(const int32_t nid = 0, const SDL_Rect& dest = {MAP_WIDTH / 2, MAP_HEIGHT / 2, BASE_WIDTH, BASE_HEIGHT}, int health = BASE_INIT_HP);
```
Comment (JohnAgapeyev) [inline src/buildings/Base.h]: Long line needs wrapping

File: src/buildings/Base.h [hunk 3]
```diff
@@ -5,28 +5,53 @@
...
     virtual ~Base();
 
     void onCollision();
     void collidingProjectile(int damage);
+    void setHealth(int newHealth) {health = newHealth;}
+    int getHealth() {return health;}
```
Comment (JohnAgapeyev) [inline src/buildings/Base.h]: const

File: src/game/GameStateMatch.cpp [hunk 4]
```diff
@@ -53,11 +52,8 @@ bool GameStateMatch::load() {
                 &GameManager::instance()->getMarine(NetworkManager::instance().getPlayerId()).first);
         GameManager::instance()->getPlayer().setId(NetworkManager::instance().getPlayerId());
     } else {
-        base.setSrcRect(BASE_SRC_X, BASE_SRC_Y, BASE_SRC_W, BASE_SRC_H);
-        GameManager::instance()->addObject(base);
-        Point newPoint = base.getSpawnPoint();
-        base.setSrcRect(BASE_SRC_X, BASE_SRC_Y, BASE_SRC_W, BASE_SRC_H);
-
+        GameManager::instance()->addObject(GameManager::instance()->getBase());
+        Point newPoint = GameManager::instance()->getBase().getSpawnPoint();
```
Comment (JohnAgapeyev) [inline src/game/GameStateMatch.cpp]: const Point

**Review by JohnAgapeyev — APPROVED** [review]
Just a few const's are missing
Otherwise it is fine

**Review by isaacmorneau — APPROVED** [review]
Same fixes as John said, but other than that, looks good

**Pull request merged**